lib/src/Uncertainty/Distribution/TruncatedDistribution.cxx

```diff
@@ -338,54 +325,72 @@ Scalar TruncatedDistribution::computeScalarQuantile(const Scalar prob,
+  Point upperBound(dimension);
+  BoolCollection finiteLowerBound(dimension);
+  BoolCollection finiteUpperBound(dimension);
+  for (UnsignedInteger k = 0; k < getDimension(); ++ k)
+  {
+    finiteLowerBound[k] = (parameter[parametersSize + k] == 1.0);
+    lowerBound[k] = parameter[parametersSize + dimension + k];
+    finiteUpperBound[k] = (parameter[parametersSize + 2 * dimension + k] == 1.0);
+    upperBound[k] = parameter[parametersSize + 3 * dimension + k];
+  }
+  Interval bounds(lowerBound, upperBound, finiteLowerBound, finiteUpperBound);
   const Scalar w = getWeight();
-  if (finiteLowerBound && finiteUpperBound) *this = TruncatedDistribution(newDistribution, lowerBound, upperBound);
-  else if (finiteLowerBound) *this = TruncatedDistribution(newDistribution, lowerBound, LOWER);
-  else if (finiteUpperBound) *this = TruncatedDistribution(newDistribution, upperBound, UPPER);
-  else throw InvalidArgumentException(HERE) << "Error: no bound";
+  *this = TruncatedDistribution(newDistribution, bounds);
   setWeight(w);
 }
 
 /* Parameters description accessor */
 Description TruncatedDistribution::getParameterDescription() const
 {
   Description description(distribution_.getParameterDescription());
-  description.add("finiteLowerBound");
-  description.add("lowerBound");
-  description.add("finiteUpperBound");
-  description.add("upperBound");
+  for (UnsignedInteger k = 0; k < getDimension(); ++ k)
+    description.add((getDimension() > 1) ? String(OSS() << "finiteLowerBound_" << k) : "finiteLowerBound");
+  for (UnsignedInteger k = 0; k < getDimension(); ++ k)
+    description.add((getDimension() > 1) ? String(OSS() << "lowerBound_" << k) : "lowerBound");
+  for (UnsignedInteger k = 0; k < getDimension(); ++ k)
+    description.add((getDimension() > 1) ? String(OSS() << "finiteUpperBound_" << k) : "finiteUpperBound");
+  for (UnsignedInteger k = 0; k < getDimension(); ++ k)
+    description.add((getDimension() > 1) ? String(OSS() << "upperBound_" << k) : "upperBound");
   return description;
 }
 
 /* Check if the distribution is elliptical */
 Bool TruncatedDistribution::isElliptical() const
 {
-  return distribution_.isElliptical() && finiteLowerBound_ && finiteUpperBound_ && (std::abs(distribution_.getRange().getLowerBound()[0] - lowerBound_ + distribution_.getRange().getUpperBound()[0] - upperBound_) < ResourceMap::GetAsScalar("Distribution-DefaultQuantileEpsilon"));
+  Bool elliptical = distribution_.isElliptical();
+  for (UnsignedInteger k = 0; elliptical && (k < getDimension()); ++ k)
+  {
+    elliptical = elliptical && bounds_.getFiniteLowerBound()[k] && bounds_.getFiniteUpperBound()[k];
+    const Scalar delta = std::abs(distribution_.getRange().getLowerBound()[k] - bounds_.getLowerBound()[k] + distribution_.getRange().getUpperBound()[k] - bounds_.getUpperBound()[k]);
+    elliptical = elliptical && (delta < ResourceMap::GetAsScalar("Distribution-DefaultQuantileEpsilon"));
+  }
+  return elliptical;
```

Comment:
Wrong.
  + For dimension==1, a distribution is elliptical if and only if it is symmetrical wrt a given point. If the underlying distribution is not elliptical, the truncated one cannot be elliptical, and if the underlying distribution is elliptical, the truncated one is elliptical if and only if the truncation interval is symmetrical wrt to symmetry point of the underlying distribution.
  + For dimension>1, a truncated distribution CANNOT be elliptical if there is an actual truncation (ie the range of the initial distribution is not included into the truncation interval), even if the underlying distribution is elliptical. It is due to the fact that the iso-PDF of an elliptical distribution are ellipses, which is not compatible with a truncation wrt a box.